tm2/pkg/db/fsdb.go

```diff
@@ -43,83 +43,98 @@ func NewFSDB(dir string) *FSDB {
 	return database
 }
 
-func (db *FSDB) Get(key []byte) []byte {
+func (db *FSDB) Get(key []byte) ([]byte, error) {
 	db.mtx.Lock()
 	defer db.mtx.Unlock()
 	key = escapeKey(key)
 
 	path := db.nameToPath(key)
 	value, err := read(path)
 	if os.IsNotExist(err) {
-		return nil
+		return nil, nil
```

Comment:
What is the idea behind returning `nil, nil`? Don't we want to handle this top-level, as the assumption is that if `Get` returns `nil`, the error should not be `nil`?

The same question is for all handlers that have this behavior

Reply:
I didn't want to change the original behavior. The contract with that interface is that if the content does not exist, it must return nil.